### PR TITLE
Feat/Contract status

### PR DESCRIPTION
**Pull request by jphinning**
In this PR, I've made the following:

- DB timezone now set to UTC, fixes the "Job Post was created 3 hours ago" in the frontend
- JobPost find by user query is now set to return in descending order
- Contracts table created, can be seen bellow 
image
- Job Post queries now return the object Contract
- After a job post is created, a contract will also be created
- Contracts endpoints were created following REST conventions, but I've also made a '/freelancer/my-contracts' and a /client/my-contracts that will be useful to list all user contracts in the near future.
- Now the frontend will be able to query contract info in the /job-post/self
image

### CODE REVIEW

**Review by xx-TOOL-xx — APPROVED** [review]

File: src/modules/contracts/contracts.controller.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  @UseGuards(JwtAuthGuard)
+  @Post()
+  create(@Body() createContractDto: CreateContractDto) {
+    return this.contractsService.create(createContractDto);
+  }
+
+  @UseGuards(JwtAuthGuard)
+  @Get()
+  findAll() {
```
Comment (lyudad) [inline src/modules/contracts/contracts.controller.ts]: return types for each functions

File: src/modules/contracts/entities/contracts.entity.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    nullable: true,
+  })
+  freelancer: Freelancer;
+
+  @Column({
+    type: 'enum',
+    enum: ContractStatus,
+    default: ContractStatus.PENDING,
+  })
+  contractStatus: ContractStatus;
```
Comment (lyudad) [inline src/modules/contracts/entities/contracts.entity.ts]: I dont think that we need status, if end date exist contract close, if end date not exist contract opened

File: src/modules/contracts/contracts.controller.ts (hunk 1 above)
Comment (jphinning) [reply src/modules/contracts/contracts.controller.ts]: Should we have explicit return types for every controller? I believe all of them are being infered as you can see below:
image

File: src/modules/contracts/entities/contracts.entity.ts (hunk 2 above)
Comment (jphinning) [reply src/modules/contracts/entities/contracts.entity.ts]: You're right, but what do you think about the logic below for the front end? 

If there's no starting date, the contract will be in the 'pending' or 'analyzing bids' state. When a proposal and job offer are accepted, a starting date will be added and the contract will be active. When the job is done, an ending date will be added and the contract will be closed

Or would you prefer only open and closed states to be shown in the front end?

File: src/modules/contracts/contracts.controller.ts (hunk 1 above)
Comment (jphinning) [reply src/modules/contracts/contracts.controller.ts]: Added explicit types for the Contracts controller 👍🏻

File: src/modules/contracts/entities/contracts.entity.ts (hunk 2 above)
Comment (jphinning) [reply src/modules/contracts/entities/contracts.entity.ts]: Removed the contract status 👍🏻

Comment (lyudad) [reply src/modules/contracts/entities/contracts.entity.ts]: I think you confused between offer and contract, for me its 2 totally different entities, as contract become from accepted offer. to I'm not fully understand how you want to work with this logic

Comment (jphinning) [reply src/modules/contracts/entities/contracts.entity.ts]: What we want right now is to display to the client in his job post information about the contract, if it is pending (no job offers accepted), open (one job offer was accepted), or if it was closed. The freelancer will also be able to see on his 'My Contracts' page all contracts that are associated with him. The job offer entity will be done later.

So in the future, once the job offer is accepted by the freelancer, the contract start date will be updated, and when everything is done or the client closes his job post, the end date will be updated.

Comment (lyudad) [reply src/modules/contracts/entities/contracts.entity.ts]: there no pending status, contract or open or close

**Pull request merged**